### PR TITLE
Add luacheck verification workflow

### PR DESCRIPTION
**Pull request by cooljeanius**
Just a simple GitHub Actions CI workflow to ensure that the lua here passes `luacheck`

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+max_line_length=140
+-- show the warning/error codes as well
+codes=true
+-- skip showing files with no issues
+quiet=1
+-- skip showing undefined variable usage
+-- there are thousands of warnings here because luacheck is unaware of Wesnoth's
```
Comment (ProditorMagnus) [inline .luacheckrc:7]: we have quite little Lua here, how it gets thousands?

Comment (cooljeanius) [reply .luacheckrc:7]: oh that's just due to the line being carried over from upstream Wesnoth's copy of the file; you can remove it if you'd like

Comment (ProditorMagnus) [reply .luacheckrc:7]: https://luacheck.readthedocs.io/en/stable/config.html#custom-sets-of-globals something smarter would make sense

Comment (cooljeanius) [reply .luacheckrc:7]: I'm not quite familiar enough with the globals used in TLU to write a custom set of them myself...

Comment (ProditorMagnus) [reply .luacheckrc:7]: wesnoth and wml should cover practically all

Comment (ProditorMagnus) [reply .luacheckrc:7]: I will use emmylua instead

Comment (cooljeanius) [reply .luacheckrc:7]: I made a new PR that ought to address this: #93

**Pull request merged**